src/OpenTelemetry.AutoInstrumentation/Logging/OtelLogging.cs

```diff
@@ -89,37 +123,53 @@ internal static long GetConfiguredFileSizeLimitBytes()
 
     private static IOtelLogger CreateLogger(string suffix)
     {
-        ISink? sink = null;
-
-        if (!ConfiguredLogLevel.HasValue)
+        if (!_configuredLogLevel.HasValue)
         {
             return NoopLogger.Instance;
         }
 
-        try
+        var sink = CreateSink(suffix);
+
+        return new InternalLogger(sink, _configuredLogLevel.Value);
+    }
+
+    private static ISink CreateSink(string suffix)
+    {
+        if (_configuredLogSink == LogSink.NoOp)
         {
-            var logDirectory = GetLogDirectory();
-            if (logDirectory != null)
+            return new NoopSink();
+        }
+        else if (_configuredLogSink == LogSink.File)
+        {
+            try
```

Comment:
Similar to the native one: ideally move this code here to a method or at least make it the last branch in the check.